feat: adding sports team project to new JS curriculum (#49135)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-sports-team-cards/641fc88d8fa7127f76e0324f.md

---
id: 641fc88d8fa7127f76e0324f
title: Step 42
challengeType: 0
dashedName: step-42
---

# --description--

Call the `setPlayerCards` function with an argument of `players.filter()`.

Inside the `filter` method, add a callback function with a parameter called `player` and implicitly return `player.nickname` is not `null`. 

# --hints--

You should call the `setPlayerCards` function and pass in `players.filter()` that filters out players without a nickname. 

```js

> assert.match(code, /\s*setPlayerCards\s*\(\s*players.filter\s*\(\s*player\s*=>\s*player.nickname\s*!==\s*null\)\s*\)\s*/)

Reviewer: ```suggestion
assert.match(code, /\s*setPlayerCards\s*\(\s*players.filter\s*\(\s*\(?\s*player\s*\)?\s*=>\s*player.nickname\s*!==\s*null\)\s*\)\s*/)
```

Optional parentheses around parameter.

> You should call the `setPlayerCards` function and pass in `players.filter()` that filters out players without a nickname. 
> 
> ```js
> 
> assert.match(code, /\s*setPlayerCards\s*\(\s*players.filter\s*\(\s*\(?\s*player\s*\)?\s*=>\s*player.nickname\s*!==\s*null\)\s*\)\s*/)
> 
> ```

Reviewer: ````suggestion
In your `nickname` case, you should call the `setPlayerCards` function.

```js
assert.match(code, /\s*case\s*('|"|`)\s*nickname\s*\1\s*:\s*setPlayerCards\(/)
```

You should pass `players.filter()` to your `setPlayerCards` call.

```js
assert.match(code, /\s*case\s*('|"|`)\s*nickname\s*\1\s*:\s*setPlayerCards\(\s*players\.filter\(/)
```

You should pass an arrow callback function to your `players.filter()` call.

```js
assert.match(code, /\s*case\s*('|"|`)\s*nickname\s*\1\s*:\s*setPlayerCards\(\s*players\.filter\(\s*\(?.*\)?\s*=>/);
```

Your `filter()` callback should take a `player` parameter.

```js
assert.match(code, /\s*case\s*('|"|`)\s*nickname\s*\1\s*:\s*setPlayerCards\(\s*players\.filter\(\s*\(?\s*player\s*\)?\s*=>/)
```

Your `filter()` callback should implicitly return the result of checking if `player.nickname` is not `null`.

```js
assert.match(code, /\s*case\s*('|"|`)\s*nickname\s*\1\s*:\s*setPlayerCards\(\s*players\.filter\(\s*\(?\s*player\s*\)?\s*=>\s*player\.nickname\s*!==\s*null\s*\)/)
```
````

More granular tests.
